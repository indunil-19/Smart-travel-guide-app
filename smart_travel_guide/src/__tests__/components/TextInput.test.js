import React from "react";
import { fireEvent, render, waitFor } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import TextInput from "../../components/TextInput";

test("should match snapshot", () => {
  const tree = renderer.create(<TextInput />).toJSON();
  expect(tree).toMatchSnapshot();
});

// const inputText = "Chris Haynes";
// const errorText = "Invalid email address";

// it("should render without crashing", () => {
//   const { getByTestId } = render(
//     <TextInput value={inputText} error={true} errorText={errorText} />
//   );
//   expect(getByTestId("text-input")).toHaveTextContent("Chirs Haynes");
// });
