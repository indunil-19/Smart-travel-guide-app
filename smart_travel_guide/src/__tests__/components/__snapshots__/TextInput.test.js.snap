// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    Object {
      "marginVertical": 12,
      "width": "100%",
    }
  }
>
  <View
    style={Object {}}
  >
    <View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 6,
            },
            Object {
              "backgroundColor": "#ffffff",
              "borderColor": "rgba(0, 0, 0, 0.54)",
              "borderRadius": 4,
              "borderWidth": 1,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "paddingBottom": 0,
            },
            Object {
              "minHeight": 64,
              "paddingTop": 8,
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholderTextColor="rgba(0, 0, 0, 0.54)"
          rejectResponderTermination={true}
          selectionColor="#2DD4BF"
          style={
            Array [
              Object {
                "flexGrow": 1,
                "margin": 0,
                "paddingHorizontal": 14,
                "zIndex": 1,
              },
              Object {
                "height": 56,
              },
              Object {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              Array [
                Object {},
              ],
            ]
          }
          testID="text-input"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
</View>
`;
